config/routes.rb

```diff
@@ -39,6 +39,15 @@
       resources :posts, only: [:index, :create, :destroy]
     end
     resources :post_types, only: [:create]
+    get '/evaluations/current', to: 'evaluations#current'
+    resources :evaluations, only: [:show, :edit, :update] do
+      resources :principles, only: [:index, :update, :create, :destroy]
+      post '/pointdetails/update', to: 'point_details#update'
```

Comment:
If it were PUT method, then it would be much more beautiful to use `resources :point_details, only: [:edit, :update]`
Why does it not the PUT method?

Reply:
It could easily be a PUT method, but that still wouldn't make it a resource.
The update and edit functions of a resource require an `:id`, but we don't necessarily have one here, because the functions are both "create or update" style.
It could be possible for the justifications resource, but since its not connected to a model, it doesn't have an id either.